src/ast_editor_state.ml

```diff
@@ -0,0 +1,109 @@
+open Import
+
+type t =
+  { (* The webview store that maps string value of Uri.t of the document to the
+       related webview. *)
+    mutable webview_map : (string, WebView.t, String.comparator_witness) Map.t
+        (* Indicates current mode inside AST explorer (true = Original, false =
+           Prerocessed) *)
+  ; mutable original_mode : bool
+        (* Contains Some value when hover mode is enabled, None otherwise *)
+  ; mutable hover_disposable : Disposable.t option
+        (* Mapping between string value of Uri.t of the original (non
+           preprocessed) document to boolean, indicating that the document has
+           changed since opening its Preprocessed Document *)
+  ; mutable pp_doc_to_changed_origin_map :
+      (string, bool, String.comparator_witness) Map.t
```

Comment:
Does this actually need to be a map? It seems like we only ever care about `false` values. Might as well make it a set then.